model/generate.go

```diff
@@ -184,6 +185,22 @@ func updateParserProject(ctx context.Context, settings *evergreen.Settings, v *V
 		return nil
 	}
 
+	// If this is the first time the parser project has been updated by a generator, cache a copy of
+	// the parser project representing its state before any generators updated it.
+	if len(pp.UpdatedByGenerators) == 0 {
+		oldPP, err := ParserProjectFindOneByID(ctx, settings, v.ProjectStorageMethod, v.Id)
+		if err != nil {
+			return errors.Wrapf(err, "finding parser project '%s' from before task generation", v.Id)
+		}
+		if oldPP == nil {
+			return errors.Errorf("parser project '%s' not found", v.Id)
+		}
+		oldPP.Id = fmt.Sprintf("%s_%s", "pre_generation", oldPP.Id)
```

Comment:
Since we're encoding a special parser project with a different ID, can we turn this `pre_generation_<id>` into a helper function that we can call where needed to get the pre-generated parser project ID?

Reply:
Added